src/main/java/org/broadinstitute/gvs/azure/cosmos/AvroReader.java

```diff
@@ -53,80 +54,138 @@ static long calculateEndLocation(ObjectNode record) {
         // insertion, or deletion. There can be multiple alts separated by commas, find the longest to compare with the
         // length of the ref to get the range of bases covered by this variant.
         long refLength = record.get("ref").asText().length();
-        Object[] sortedAltLengths = Arrays.stream(record.get("alt").asText().split(",")).map(String::length).sorted().toArray();
-        int maxAltLength = (Integer) sortedAltLengths[sortedAltLengths.length - 1];
+
+        @SuppressWarnings("OptionalGetWithoutIsPresent")
+        int maxAltLength = Arrays.stream(record.get("alt").asText().split(",")).map(String::length).max(Integer::compareTo).get();
 
         return location.asLong() + Math.max(refLength, maxAltLength) - 1;
     }
 
+    private static void finishCurrentDocument(ObjectNode currentDocument, long currentMaxLocation, String dropState) {
+        ObjectNode location = (ObjectNode) currentDocument.get("location");
+        location.set("end", new LongNode(currentMaxLocation));
+        if (dropState != null) {
+            currentDocument.put("dropState", dropState);
+        }
+    }
+
     @VisibleForTesting
-    static List<ObjectNode> objectNodesForAvroPath(
-            ObjectMapper objectMapper, Path path, IngestArguments ingestArguments, AtomicLong id, AtomicLong counter) {
+    static List<ObjectNode> documentsForAvroPath(
+            ObjectMapper objectMapper, Path avroPath, IngestArguments ingestArguments,
+            AtomicLong recordCounter, AtomicLong documentCounter) {
         // Where the Cosmos JSON serialization magic happens:
         // https://github.com/Azure/azure-sdk-for-java/blob/80b12e48aeb6ad2f49e86643dfd7223bde7a9a0c/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/JsonSerializable.java#L255
-        //
 
-        File file = new File(path.toString());
+        File avroFile = new File(avroPath.toString());
         GenericDatumReader<?> reader = new GenericDatumReader<>();
-        long currentMaxLocation = -1L;
         List<ObjectNode> documentList = new ArrayList<>();
         ObjectNode currentDocument = null;
-        ArrayNode currentArrayNode = null;
+        ArrayNode currentRecordArray = null;
         Long currentSampleId = null;
+        long currentMaxLocation = -1L;
+        short currentChromosome = -1;
+        ArrayNode avroSchema = null;
+        // Why is this not a number in BQ?
```

Comment:
What is this comment referring to?

Reply:
Drop state on the next line. However Rori answered this question in mobbing (so we can store weird states that are not Phred-scaled qualities using letters), so this comment can go.